Statiq.Helpers/SocialImages.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Playwright;
+using Statiq.Common;
+using Statiq.Core;
+using Statiq.Html;
+using Statiq.Web;
+using Statiq.Web.Modules;
+using Statiq.Web.Pipelines;
+
+namespace Thirty25.Statiq.Helpers
+{
+    public class SocialImages : Pipeline
+    {
+        public SocialImages()
+        {
+            Dependencies.AddRange(nameof(Inputs), nameof(Data));
```

Comment:
You (probably) don't need the dependency on the `Data` pipeline here. The flow in Statiq Web is for the `Input` pipeline to essentially read _everything_ and determine if it's content, data, etc. by setting the correct `ContentType`. The `input` pipeline also does some universal processing like loading front matter and sidecar files. So by setting a dependency on the `Input` pipeline, you're basically positioned after all the site files have been read, front matter/sidecar files processed, and content type set.

The question is where the outputs from this pipeline are going to be used. You might need to set `DependencyOf` in this pipeline to make sure that pipeline can read the outputs. In other words, if these social cards need to be matched up with their corresponding Markdown or Razor files, you'll want to set (something like, this is from memory and may not be quite right) `DependencyOf.Add(nameof(Content))`. On the other hand, if this pipeline just needs to output the social card images to a defined, known location and you don't need to actually _use_ the output documents in the content files, there's no reason to set the extra dependency.